### PR TITLE
Fix DROOLS-2380 DROOLS-2381

### PR DESCRIPTION
**Pull request by tarilabs**
Provides fixes for 2 JIRA in https://github.com/kiegroup/drools/pull/1810

### CODE REVIEW

File: kie-dmn/kie-dmn-backend/src/main/java/org/kie/dmn/backend/marshalling/v1_1/xstream/extensions/DecisionServicesConverter.java [hunk 1]
```diff
@@ -19,12 +19,12 @@
...
 import com.thoughtworks.xstream.io.HierarchicalStreamWriter;
-import org.kie.dmn.backend.marshalling.v1_1.xstream.DMNElementConverter;
+import org.kie.dmn.backend.marshalling.v1_1.xstream.DMNModelInstrumentedBaseConverter;
 import org.kie.dmn.model.v1_1.DMNModelInstrumentedBase;
 import org.kie.dmn.model.v1_1.DecisionService;
-import org.kie.dmn.model.v1_1.DecisionServices;
+import org.kie.dmn.model.v1_1.extensions.DecisionServices;
 
-public class DecisionServicesConverter extends DMNElementConverter {
+public class DecisionServicesConverter extends DMNModelInstrumentedBaseConverter {
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-backend/src/main/java/org/kie/dmn/backend/marshalling/v1_1/xstream/extensions/DecisionServicesConverter.java:27]: DecisionServices with the ending S is a drools-only extension, more below.

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/assembler/DMNAssemblerService.java [hunk 2]
```diff
@@ -93,12 +95,13 @@ public void addResource(Object kbuilder, Resource resource, ResourceType type, R
     }
 
     private List<DMNProfile> getDMNProfiles(KnowledgeBuilderImpl kbuilderImpl) {
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/assembler/DMNAssemblerService.java:97]: Assembler code modified to be the unique source of entrance for getting "default profiles" and the likes, as discussed.

File: kie-dmn/kie-dmn-model/src/main/java/org/kie/dmn/model/v1_1/extensions/DecisionServices.java [hunk 3]
```diff
@@ -13,12 +13,15 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.kie.dmn.model.v1_1;
+package org.kie.dmn.model.v1_1.extensions;
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-model/src/main/java/org/kie/dmn/model/v1_1/extensions/DecisionServices.java:16]: DecisionServices with the ending S is not part of the original DMN v1.1 spec.

File: kie-dmn/kie-dmn-model/src/main/java/org/kie/dmn/model/v1_1/extensions/DecisionServices.java [hunk 4]
```diff
@@ -13,12 +13,15 @@
...
 
 import java.util.ArrayList;
 import java.util.List;
 
-public class DecisionServices extends NamedElement {
+import org.kie.dmn.model.v1_1.DMNModelInstrumentedBase;
+import org.kie.dmn.model.v1_1.DecisionService;
+
+public class DecisionServices extends DMNModelInstrumentedBase {
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-model/src/main/java/org/kie/dmn/model/v1_1/extensions/DecisionServices.java:24]: it is NOT a NamedElement, just a part of the "model" pojo

File: kie-dmn/kie-dmn-validation/src/main/java/org/kie/dmn/validation/DMNValidatorFactory.java [hunk 5]
```diff
@@ -16,10 +16,19 @@
...
+import java.util.Collections;
+import java.util.List;
+
+import org.kie.dmn.core.compiler.DMNProfile;
+
 public final class DMNValidatorFactory {
 
     public static DMNValidator newValidator() {
-        return new DMNValidatorImpl();
+        return new DMNValidatorImpl(Collections.emptyList());
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-validation/src/main/java/org/kie/dmn/validation/DMNValidatorFactory.java:27]: by default does not supply any user-provided Profiles

File: kie-dmn/kie-dmn-validation/src/main/java/org/kie/dmn/validation/DMNValidatorImpl.java [hunk 6]
```diff
@@ -98,6 +115,10 @@ public DMNValidatorImpl() {
             String message = MsgUtil.createMessage( Msg.FAILED_VALIDATOR );
             failedInitMsg.add(new DMNMessageImpl(DMNMessage.Severity.ERROR, message, Msg.FAILED_VALIDATOR.getType(), null ) );
         }
+        ChainedProperties localChainedProperties = new ChainedProperties();
+        this.dmnProfiles.addAll(DMNAssemblerService.getDefaultDMNProfiles(localChainedProperties));
+        this.dmnProfiles.addAll(dmnProfiles);
+        this.dmnCompilerConfig = DMNAssemblerService.compilerConfigWithKModulePrefs(localChainedProperties, this.dmnProfiles);
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-validation/src/main/java/org/kie/dmn/validation/DMNValidatorImpl.java:121]: the compiler configuration, used below when instantiating the compiler, is defined by:
- checking local ChainedProperties (at the moment we don't support user providing this externally)
- any default profiles, as descending from the refactored code on DMNAssembler
- any user-supplied profiles

**Review by etirelli — APPROVED** [review]

**Pull request merged**